test/src/test/java/org/corfudb/runtime/collections/DiskBackedCorfuClientTest.java

```diff
@@ -0,0 +1,497 @@
+        resetTests();
+        try (final CorfuTable<String, String> table = setupTable()) {
+            executeTx(() -> intended.forEach(value -> table.put(value, value)));
+            Assertions.assertEquals(table.size(), intended.size());
+            executeTx(() -> {
+                intended.forEach(value -> Assertions.assertEquals(table.get(value), value));
+                intended.forEach(table::remove);
+            });
+
+            executeTx(() -> {
+                final Set<String> persisted = new HashSet<>(table.scanAndFilter(entry -> true));
+                Assertions.assertTrue(persisted.isEmpty());
+            });
+        }
+    }
+
+    /**
+     * Non-transactional property based test that does inserts followed by removes.
+     */
+    @Property(tries = NUM_OF_TRIES)
+    void nonTxInsertRemove(@ForAll @Size(SAMPLE_SIZE) Set<String> intended) {
+        resetTests();
+        try (final CorfuTable<String, String> table = setupTable()) {
+            executeTx(() -> intended.forEach(value -> table.insert(value, value)));
+            Assertions.assertEquals(table.size(), intended.size());
+            executeTx(() ->  intended.forEach(table::delete));
+
+            executeTx(() -> {
+                final Set<String> persisted = new HashSet<>(table.scanAndFilter(entry -> true));
+                Assertions.assertTrue(persisted.isEmpty());
+            });
+        }
+    }
+
+    /**
+     * Transactional property based test that does inserts followed by removes.
+     */
+    @Property(tries = NUM_OF_TRIES)
+    void txInsertRemove(@ForAll @Size(SAMPLE_SIZE) Set<String> intended) {
+        resetTests();
+        try (final CorfuTable<String, String> table = setupTable()) {
+            intended.forEach(value -> table.insert(value, value));
+            Assertions.assertEquals(table.size(), intended.size());
+            intended.forEach(table::delete);
+
+            executeTx(() -> {
+                final Set<String> persisted = new HashSet<>(table.scanAndFilter(entry -> true));
+                Assertions.assertTrue(persisted.isEmpty());
+            });
+        }
+    }
+
+    /**
+     * Non-transactional property based test that does inserts followed by clear.
+     */
+    @Property(tries = NUM_OF_TRIES)
+    void nonTxInsertClear(@ForAll @Size(SAMPLE_SIZE) Set<String> intended) {
+        resetTests();
+        try (final CorfuTable<String, String> table = setupTable()) {
+            intended.forEach(value -> table.insert(value, value));
+            Assertions.assertEquals(table.size(), intended.size());
+            table.clear();
+
+            final Set<String> persisted = new HashSet<>(table.scanAndFilter(entry -> true));
+            Assertions.assertEquals(persisted.size(), 0);
+            Assertions.assertEquals(table.size(), 0);
+        }
+    }
+
+    /**
+     * Transactional property based test that does inserts followed by clear.
+     */
+    @Property(tries = NUM_OF_TRIES)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use explicit scoping instead of the default package private level](https://app.codacy.com/app/annymartinez.m/CorfuDB/pullRequest?prid=4651451)